src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs

```diff
@@ -432,6 +432,7 @@ public static ImmutableArray<TSource> ToImmutableArray<TSource>(this ImmutableAr
         [Pure]
         public static int BinarySearch<T>(this ImmutableArray<T> array, T value)
         {
+            Debug.Assert(array.array != null);
```

Comment:
As I know, `ImmutableArray` is intended to throw NRE or let called method throw when uninitialized. It doesn't provide any null check itself. Asserts should not be used here.

Reply:
oh yeah I should have banged it

Reply:
I asserted it because it is public APIs and nothing mentioned in the doc about NRE.